src/consent-manager.js

```diff
@@ -160,76 +166,94 @@ export default class ConsentManager {
         }
 
         const elements = document.querySelectorAll("[data-name='"+app.name+"']")
-        for(var i=0;i<elements.length;i++){
-            const element = elements[i]
 
+        var i, k, l, m, attrs, attr, val
+
+        for (i = 0, l = elements.length; i < l; i++) {
+            const element = elements[i]
             const parent = element.parentElement
-            const {dataset} = element
-            const {type, name} = dataset
-            const attrs = ['href', 'src']
+            const dataset = getDataAttrs(element)
 
-            //if no consent was given we disable this tracker
-            //we remove and add it again to trigger a re-execution
+            // if no consent was given we disable this tracker
+            // we remove and add it again to trigger a re-execution
 
-            if (element.tagName === 'SCRIPT'){
+            if (element.tagName === 'SCRIPT') {
                 // we create a new script instead of updating the node in
                 // place, as the script won't start correctly otherwise
                 const newElement = document.createElement('script')
-                for(var key of Object.keys(dataset)){
-                    newElement.dataset[key] = dataset[key]
+
+                for (k in dataset) {
+                    if (dataset.hasOwnProperty(k)) {
+                        newElement.setAttribute(k, dataset[k])
+                    }
                 }
                 newElement.type = 'opt-in'
-                newElement.innerText = element.innerText
-                newElement.text = element.text
-                newElement.class = element.class
                 newElement.style.cssText = element.style
-                newElement.id = element.id
-                newElement.name = element.name
-                newElement.defer = element.defer
-                newElement.async = element.async
-
-                if (consent){
-                    newElement.type = type
-                    if (dataset.src !== undefined)
-                        newElement.src = dataset.src
+
+                attrs = 'innerText text class id name defer async charset'.split(' ');
+                for (k = 0, m = attrs.length; k < m; k++) {
```

Comment:
Please don't reuse variables for different purposes (`k` in this case). This makes the code harder to read.